tests/drivers/rtc/rtc_api/testcase.yaml

```diff
@@ -1,13 +1,25 @@
 # Copyright (c) 2022 Bjarki Arge Andreasen
+# Copyright (c) 2024 STMicroelectronics
 # SPDX-License-Identifier: Apache-2.0
 
+common:
+  tags:
+    - drivers
+    - rtc
+    - api
+  filter: dt_alias_exists("rtc")
+  depends_on: rtc
 tests:
   drivers.rtc.rtc_api:
-    tags:
-      - drivers
-      - rtc
-      - api
-    filter: dt_alias_exists("rtc")
-    depends_on: rtc
     platform_exclude:
       - qemu_x86_64
+  drivers.rtc.rtc_api.stm32:
+    filter: CONFIG_SOC_FAMILY_STM32
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/stm32-rtc.overlay"
+    platform_allow:
+      - b_u585i_iot02a
+      - nucleo_f767zi
+      - nucleo_h563zi
+      - nucleo_h723zg
+      - nucleo_u575zi_q
```

Comment:
Board specific modifications should be in `boards/` rather than the testcase.yaml

Using `boards/*.overlay` and `boards/*.conf`, modifications to Kconfig and dts is possible.